src/dr_llm/logging/events.py

```
@@ -20,27 +39,44 @@ class GenerationLogEvent(BaseModel):
+    thread_id: int = Field(default_factory=get_ident)
     payload: dict[str, Any] = Field(default_factory=dict)
 
+    @classmethod
+    def from_context(
+        cls,
+        *,
+        event_type: str,
+        stage: str,
+        payload: dict[str, Any],
+        context: Mapping[str, Any],
+    ) -> "GenerationLogEvent":
```

Ruff (0.15.9)

[warning] Remove quotes from type annotation

Remove quotes

(UP037)